staging/src/k8s.io/client-go/rest/transport.go

```diff
@@ -94,7 +95,15 @@ func (c *Config) TransportConfig() (*transport.Config, error) {
 	}
 
 	if c.ExecProvider != nil {
-		provider, err := exec.GetAuthenticator(c.ExecProvider)
+		var cluster clientauthentication.Cluster
```

Comment:
make this a pointer, don't dereference, and pass a pointer to GetAuthenticator?

```suggestion
		var cluster *clientauthentication.Cluster
```

Reply:
I agree this would clean up the code here in `transport.go`. I was trying to avoid passing a pointer to `GetAuthenticator` that could be nil or non-nil, so that `GetAuthenticator` wouldn't need to worry about limping around with a nil pointer. On the other hand, passing this struct by pointer would alleviate some copying of a `Cluster` up the stack. I will go ahead and update the `GetAuthenticator` code, and friends, to take a pointer. It doesn't look like we touch that `cluster` struct very much right now.

Reply:
bfe3c622358